# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/painting/nsDisplayList.h
# Commit: 13f70dfa4e1d
# Full Hash: 13f70dfa4e1d1d2c0f00b851ca655529f624fddd
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2021-05-16 09:17:48
# Regressor Bug: 1540737
# File Overlap Count: 1
# Description:
#   Bug 1540737 - Allow filter painting to take a callback for painting children. r=miko
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114670
# ==============================================================================

diff -r 10dacf71ccbe -r 13f70dfa4e1d layout/painting/nsDisplayList.h
--- a/layout/painting/nsDisplayList.h	Sun May 16 02:28:40 2021 +0300
+++ b/layout/painting/nsDisplayList.h	Sun May 16 03:55:29 2021 +0000
@@ -6649,6 +6649,10 @@
   void PaintAsLayer(nsDisplayListBuilder* aBuilder, gfxContext* aCtx,
                     LayerManager* aManager);
 
+  void PaintWithContentsPaintCallback(
+      nsDisplayListBuilder* aBuilder, gfxContext* aCtx,
+      const std::function<void(gfxContext* aContext)>& aPaintChildren);
+
   bool CreateWebRenderCommands(
       mozilla::wr::DisplayListBuilder& aBuilder,
       mozilla::wr::IpcResourceUpdateQueue& aResources,